src/Microsoft.AspNetCore.Mvc.ViewFeatures/Rendering/TagBuilder.cs

```diff
@@ -57,7 +57,23 @@ public AttributeDictionary Attributes
         /// <summary>
         /// Gets the inner HTML content of the element.
         /// </summary>
-        public IHtmlContentBuilder InnerHtml { get; }
+        public IHtmlContentBuilder InnerHtml
+        {
+            get
+            {
+                if (_innerHtml == null)
+                {
+                    _innerHtml = new HtmlContentBuilder();
+                }
+
+                return _innerHtml;
+            }
+        }
+
+        /// <summary>
+        /// Gets an indication <see cref="InnerHtml"/> has been access and is likely not empty.
+        /// </summary>
+        public bool HasInnerHtml => _innerHtml != null;
```

Comment:
This is a public property on a public class. It surprising behavior to have it return a value that differs based on whether or not another property was **read**